drivers/scheduler/k8s/k8s.go

```diff
@@ -2929,6 +2936,55 @@ func (k *K8s) getExpectedAutopilotSizeForPvc(pvc *v1.PersistentVolumeClaim, apRu
 	return expectedPVCSize, nil
 }
 
+// ValidateEvents verifies proper alerts and events on resize completion
+func (k *K8s) ValidateEvents(pvc *v1.PersistentVolumeClaim, apRules *apapi.AutopilotRuleList) error {
+
+	// wait until autopilot posts all events
+	time.Sleep(time.Minute)
+
+	successEvens := []string{
+		"Initializing => Normal",
+		"Normal => Triggered",
+		"Triggered => ActiveActionsPending",
+		"ActiveActionsPending => ActiveActionsInProgress",
+		"ActiveActionsInProgress => ActiveActionsTaken",
+	}
+
+	for _, apRule := range apRules.Items {
+		if isAutopilotMatchPvcLabels(apRule, pvc) {
+
+			fields := fmt.Sprintf("involvedObject.kind=%s,involvedObject.name=%s", "AutopilotRule", apRule.Name)
+			events, err := k8sops.Instance().ListEvents("", metav1.ListOptions{FieldSelector: fields})
+			if err != nil {
+				return err
+			}
+
+			eventMap := make(map[string]bool)
+			for _, event := range events.Items {
+				if time.Since(event.CreationTimestamp.Time).Hours() > float64(1) {
+					logrus.Infof("Can't capture all events for %s:%s, since resize has started more that 1h ago", apRule.Name, pvc.Name)
+					return nil
+				}
+
+				// cut last 3 entries from a message like `rule: pvc:pvc-25c transition from ActiveActionsInProgress => ActiveActionsTaken`
+				transitionMsg := strings.Split(event.Message, " ")
+				eventMap[strings.Join(transitionMsg[len(transitionMsg)-3:], " ")] = true
+			}
+			return checkEvents(successEvens, eventMap)
+		}
+	}
+	return nil
+}
+
+func checkEvents(successEvens []string, eventMap map[string]bool) error {
+	for _, expectedEvent := range successEvens {
+		if _, ok := eventMap[expectedEvent]; !ok {
+			return fmt.Errorf("failed to validate events. Expected %s", expectedEvent)
+		}
+	}
+	return nil
```

Comment:
add logic where  ActiveActionsTaken => Normal is the last event for PVC and no more after that, otherwise it will pass here even if it has Normal => Triggered after that